packages/react-instantsearch/src/core/utils.js

```diff
@@ -64,14 +64,12 @@ export const defer = f => {
 };
 
 export function removeEmptyKey(obj) {
-  Object.keys(obj).forEach(
-    key =>
-      (obj[key] &&
-        typeof obj[key] === 'object' &&
-        !isEmpty(obj[key]) &&
-        removeEmptyKey(obj[key])) ||
-      obj[key] === undefined ||
-      (isEmpty(obj[key]) && delete obj[key])
-  );
+  each(obj, (v, k) => {
+    if (isEmpty(v) && isObject(v)) {
+      delete obj[k];
+    } else if (isObject(v)) {
```

Comment:
* isObject: Checks if value is the language type of Object. (e.g. arrays, functions, objects, regexes, new Number(0), and new String(''))
* isPlainObject: Checks if value is a plain object, that is, an object created by the Object constructor or one with a [[Prototype]] of null.

So most of the things are Objects, but in this case we only want "real" objects (maps)